### DISCUSSION TITLE
Ability to add Panes in module?

### DISCUSSION THREAD

**Discussion by Mamik153** [oqtane/oqtane.framework#2216] [Q&A]
Hi, 

I was wondering if it was possible to define panes in a module and then place like other module inside those panes. Is it possible or has anyone tried something similar? 
The basic idea is this:

- I have a module that defines a structure like maybe a 3 column grid
- I place that module inside a pane defined in Theme
- The place other modules inside the panes defined inside the module

Thank you

**Comment by sbwalker** [comment]
That is currently not possible. The framework enforces a clear separation of concerns - modules are for managing data/content, and themes are for managing layout/appearance.

**Reply by Mamik153** [reply]
Okay understood thanks 👍

Is there any plan for having the ability to do this in the near future? Getting curious now 😶

**Reply by sbwalker** [reply]
There are no plans to add this capability to the framework as it would break the clean separation of concerns. You could certainly create a custom theme with the layout you need and apply it to the page where the module resides. You could also create a custom module if you need something more specific for your use case.

**Reply by Mamik153** [reply]
Thank you for the info
